packages/typespec-ts/test/modularIntegration/generated/models/inheritance/enum-discriminator/src/models/models.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/** Test extensible enum type for discriminator */
+export interface DogParent {
+  /** the discriminator possible values golden */
+  kind: string;
+  /** Weight of the dog */
+  weight: number;
+}
+
+/** Golden dog model */
+export interface Golden extends DogParent {
+  /** discriminator property */
+  kind: "golden";
+}
+
+/** extensible enum type for discriminator */
+/** "golden" */
+export type DogKind = string;
+
+/** Test fixed enum type for discriminator */
+export interface SnakeParent {
+  /** the discriminator possible values cobra */
+  kind: string;
```

Comment:
Is this kind a fixed enum? Will we treat the same as extensible and take as string?

Reply:
Yes, see the point 4 of this issue https://github.com/Azure/autorest.typescript/issues/2143